x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -168,13 +168,13 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
       config: (): Promise<ConfigType> => Promise.resolve(config),
     };
 
-    const router = core.http.createRouter();
-    core.http.registerRouteHandlerContext(
-      APP_ID,
-      (context, request, response): AppRequestContext => ({
-        getAppClient: () => this.appClientFactory.create(request),
-      })
-    );
+    const router = core.http.createRouter<SecuritySolutionRequestHandlerContext>();
+    core.http.registerRouteHandlerContext<
+      SecuritySolutionRequestHandlerContext,
+      'securitySolution'
```

Comment:
would using `APP_ID` here this way work?

```suggestion
      typeof APP_ID
```

Reply:
updated